template/memory-gui.js

```diff
@@ -1,29 +1,99 @@
-var MemoryGUI = (function () {
-
-	//...
-
-	function GuiCtor(container,game) {
-		game.gui(this); // link game to this gui
-
-		//...
-
-		// public instance methods
-		// (you may instead attach these to a prototype if you prefer)
-		this.reset = function() {
-			//...
-		}
-		this.show = function(where,displayString) {
-			//...
-		}
-		this.removeSoon = function(whereArr) {
-			//...
-		}
-		this.hideSoon = function(whereArr) {
-			//...
-		}
-
-		// Do some initial setup and rendering...
-	}
-
-	return GuiCtor;
+var MemoryGUI = (function() {
```

Comment:
Your GUI is working great! Code is laid out appropriately and is functioning. Your GUI is just taking directions from the game's logic and not making its own decisions, which is perfect.

Some enhancements you can make:

It's currently possible to have more than two cards showing at any given time before your hideSoon function triggers. Try playing with the value for setTimeout a bit or maybe just disabling the click function while two cards are showing. This is a small issue, but also an example that there are always improvements to be made.

Overall, great work!